api/v1alpha1/hostedcluster_types.go

```diff
@@ -152,6 +152,23 @@ type PlatformSpec struct {
 	// AWS contains AWS-specific settings for the HostedCluster
 	// +optional
 	AWS *AWSPlatformSpec `json:"aws,omitempty"`
+
+	// IBMCloud contains IBMCloud-specific settings for the HostedCluster
+	// +optional
+	IBMCloud *IBMCloudPlatformSpec `json:"ibmcloud,omitempty"`
+}
+
+type IBMCloudPlatformSpec struct {
+	// AuditWebhook contains metadata for configuring an audit webhook
+	// endpoint for a cluster
+	// +optional
+	AuditWebhook *AuditWebhook `json:"auditWebhook,omitempty"`
+}
+
+type AuditWebhook struct {
+	// Enable enables the audit webhook
+	// +optional
+	Enable bool `json:"enable,omitempty"`
 }
```

Comment:
Given this is a k8s/OCP feature and not a cloud provider feature, I propose moving `AuditWebhook` to `HostedCluster` itself as an optional field with documentation about which cloud platforms currently support the feature (IBMCloud in this case).

In addition, let's consider validating the feature's platform support, and if the feature is enabled on an unsupported platform, factoring that into the HostedControlPlane Available status condition calculation (perhaps via KAS status) to ensure the HostedControlPlane is surfaced as misconfigured.

Validating at admission time with a webhook is yet another layer we could add.